### PR TITLE
buildFHSUserEnv{Chroot,Bubblewrap}: link gsettings-schemas to the FHS…

### PR DESCRIPTION
**Pull request by Artturin**
… location

###### Motivation for this change
Supersedes  #83705

###### Things done
Tested by launching with `XDG_DATA_DIRS= ./result/bin/ssb-patchwork` and selecting a profile picture which opens a file picker

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/build-fhs-userenv-bubblewrap/env.nix [hunk 1]
```diff
@@ -93,6 +93,28 @@ let
...
+      if [[ -d  $out/share/gsettings-schemas/ ]]; then
+          # Recreate the standard schemas directory if its a symlink to make it writable
+          if [[ -L $out/share/glib-2.0 ]]; then
+              ln -s $(readlink $out/share/glib-2.0) $out/share/glib-2.0.old
+              rm -rf $out/share/glib-2.0
+              mkdir -p $out/share/glib-2.0/schemas
+          fi
+
+          # symlink any schema files or overrides to the standard schema directory
+          ln -fs $out/share/glib-2.0.old/schemas/*.xml $out/share/glib-2.0/schemas
```
Comment (wentasah) [inline pkgs/build-support/build-fhs-userenv-bubblewrap/env.nix:106]: If `$out/share/glib-2.0` is not a symlink, `$out/share/glib-2.0.old` won't exist and this command either fails (if `nullopt` is unset) or creates a symlink to `$out/share/glib-2.0/schemas` in the current directory (with `nullopt` set).

Comment (Artturin) [reply pkgs/build-support/build-fhs-userenv-bubblewrap/env.nix]: fixed, thanks for the review

**Comment by Artturin** [comment]
moved `mkdir -p $out/share/glib-2.0/schemas` out of a if block to make sure it always exists

**Comment by Artturin** [comment]
TODO remove `gsettings-desktop-schemas` and `hicolor-icon-theme`  from packages that use the appimage builder

Result of `nixpkgs-review pr 161739` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>ovftool</li>
    <li>timedoctor</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>appimage-run-tests</li>
  </ul>
</details>
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>ciscoPacketTracer7</li>
    <li>ciscoPacketTracer8</li>
    <li>houdini</li>
    <li>quartus-prime-lite</li>
    <li>runescape</li>
  </ul>
</details>
<details>
  <summary>109 packages built:</summary>
  <ul>
    <li>Sylk</li>
    <li>altair</li>
    <li>android-studio (androidStudioPackages.stable)</li>
    <li>androidStudioPackages.beta</li>
    <li>androidStudioPackages.canary</li>
    <li>androidStudioPackages.dev</li>
    <li>anki-bin</li>
    <li>anytype</li>
    <li>appimage-run</li>
    <li>apple-music-electron</li>
    <li>arduino</li>
    <li>arduino-ci</li>
    <li>arduino-cli</li>
    <li>arduino-core</li>
    <li>beekeeper-studio</li>
    <li>betterdiscord-installer</li>
    <li>beyond-identity</li>
    <li>bitscope.chart</li>
    <li>bitscope.console</li>
    <li>bitscope.display</li>
    <li>bitscope.dso</li>
    <li>bitscope.logic</li>
    <li>bitscope.meter</li>
    <li>bitscope.proto</li>
    <li>bitscope.server</li>
    <li>bloomrpc</li>
    <li>bottles</li>
    <li>buttercup-desktop</li>
    <li>chrysalis</li>
    <li>clonehero</li>
    <li>conda</li>
    <li>cozy-drive</li>
    <li>crypto-org-wallet</li>
    <li>devdocs-desktop</li>
    <li>dropbox</li>
    <li>dropbox-cli</li>
    <li>electron-mail</li>
    <li>electronplayer</li>
    <li>esphome</li>
    <li>fahclient</li>
    <li>firefly-desktop</li>
    <li>flexoptix-app</li>
    <li>flutter (flutterPackages.stable)</li>
    <li>fspy</li>
    <li>heroic</li>
    <li>hover</li>
    <li>irccloud</li>
    <li>itch</li>
    <li>itch-setup</li>
    <li>joplin-desktop</li>
    <li>keeweb</li>
    <li>kodiPackages.steam-launcher</li>
    <li>lbry</li>
    <li>ledger-live-desktop</li>
    <li>left4gore-bin</li>
    <li>lens</li>
    <li>lightworks</li>
    <li>losslesscut-bin</li>
    <li>lunar-client</li>
    <li>lutris</li>
    <li>lutris-free</li>
    <li>marktext</li>
    <li>mate.caja-dropbox</li>
    <li>mathpix-snipping-tool</li>
    <li>minigalaxy</li>
    <li>mobilecoin-wallet</li>
    <li>molotov</li>
    <li>mycrypto</li>
    <li>neo4j-desktop</li>
    <li>notable</li>
    <li>notion-app-enhanced</li>
    <li>nuclear</li>
    <li>octant-desktop</li>
    <li>p3x-onenote</li>
    <li>pdfstudio</li>
    <li>pdfstudioviewer</li>
    <li>platformio</li>
    <li>playonlinux</li>
    <li>plex</li>
    <li>plexamp</li>
    <li>protontricks</li>
    <li>radicle-upstream</li>
    <li>rambox-pro</li>
    <li>saleae-logic-2</li>
    <li>session-desktop-appimage</li>
    <li>shticker-book-unwritten</li>
    <li>sidequest</li>
    <li>ssb-patchwork</li>
    <li>standardnotes</li>
    <li>station</li>
    <li>steam</li>
    <li>steam-run</li>
    <li>steam-tui</li>
    <li>steamcmd</li>
    <li>teensyduino</li>
    <li>timeular</li>
    <li>trezor-suite</li>
    <li>tusk</li>
    <li>unityhub</li>
    <li>unvanquished</li>
    <li>via</li>
    <li>vial</li>
    <li>vmware-horizon-client</li>
    <li>vscode-fhs</li>
    <li>vscodium-fhs</li>
    <li>wootility</li>
    <li>ytmdesktop</li>
    <li>zettlr</li>
    <li>zulip</li>
  </ul>
</details>

**Pull request merged**